gslib/utils/copy_helper.py

```diff
@@ -2196,6 +2199,9 @@ def CallResumableUpload():
                               provider=dst_url.scheme)
       raise
 
+  if using_temporary_file:
```

Comment:
Can we move this outside of this method? I think this kinda break the contract as this method is just supposed to upload the file and not delete the source. Maybe the caller can decide to delete the source.

Reply:
It seems more awkward to move the delete to `PerformCopy` to me:
Before:
```
if dst_url.IsCloudUrl():
      return _UploadFileToObject(src_url,
                                 src_obj_filestream,
                                 ...
                                 using_temporary_file=use_stet)
```
After:
```
if dst_url.IsCloudUrl():
      uploaded_metadata = _UploadFileToObject(src_url,
                                 src_obj_filestream,
                                 ...
                                 using_temporary_file=use_stet)
      if use_stet:
        # Delete temporary file.
        os.unlink(src_obj_filestream.name)
```
It breaks a different pattern in `PerformCopy` where we have:
```
if [identify copy type]:
  return [copy method]
```

Reply:
Hmm, I would prefer that if it means keeping the job of private methods limited to what they are meant to do. Plus you also avoid passing the`use_stet` flag.

Regarding the pattern in performCopy, I think breaking a pattern is better than breaking a contract.

Having said that, I am not strongly opposed to having it in UploadFileToObject so I am fine if other reviewers are okay with it.

Reply:
Is it really a contract? `DownloadObjectToFile` calls `ValidateAndCompleteDownload`, which deletes temporary files

Reply:
I meant `DownloadObjectToFile` b/c it's another example of an "execute copy" function `PerformCopy` calls.

But moved the delete to `PerformCopy` so that it handles temp file creation and deletion.